### PR TITLE
test-run: bump to new version

### PR DESCRIPTION
**Pull request by kyukhin**
Bump test-run to new version with the following improvements:
- Add timeout for starting tarantool server (tarantool/test-run#302)
- Kill hanging processes of not started servers (tarantool/test-run#332)
- Rerun all failed tests, not only marked as fragile (tarantool/test-run#329)

Also, set tarantool server start timeout to 290 secs and exclude test-run
tests from validation by luacheck.

NO_DOC=testing
NO_TEST=testing
NO_CHANGELOG=testing

### CODE REVIEW

File: .luacheckrc [hunk 1]
```diff
@@ -31,6 +31,7 @@ include_files = {
 exclude_files = {
     "build/**/*.lua",
     "test-run/test/test-tarantool/*.test.lua",
+    "test-run/test/unittest/*.test.lua",
```
Comment (ylobankov) [inline .luacheckrc:34]: I think running luacheck against a git submodule is redundant. The test-run submodule itself has the own CI where all checks are run including luacheck. So my proposal is the following: replace all test-run stuff ->  `"test-run/**/*.lua"`. And please move this change to a separate commit.

File: .luacheckrc [hunk 2]
```diff
@@ -31,6 +31,7 @@ include_files = {
 exclude_files = {
```
[the remark this answers is not in the document]
Comment (kyukhin) [reply .luacheckrc:31]: Done both.

File: .luacheckrc (hunk 1 above)
Comment (kyukhin) [reply .luacheckrc:34]: Done. However, there was no need for separate commit [before](https://github.com/tarantool/tarantool/commit/8b3ec7b928a02a8cc44aca1911cc5d665cf5133d).

File: .luacheckrc [hunk 3]
```diff
@@ -30,7 +30,7 @@ include_files = {
 
 exclude_files = {
     "build/**/*.lua",
-    "test-run/test/test-tarantool/*.test.lua",
+    "test-run/**//*.test.lua",
     "test-run/lib/checks/*",
```
Comment (ylobankov) [inline .luacheckrc:34]: This and the next line can be deleted.

File: .github/actions/environment/action.yml [hunk 4]
```diff
@@ -5,6 +5,7 @@ runs:
   steps:
     - run: |
         echo VARDIR=/tmp/tnt | tee -a $GITHUB_ENV
+        echo SERVER_START_TIMEOUT=290 | tee -a $GITHUB_ENV
```
Comment (ylobankov) [inline .github/actions/environment/action.yml:8]: Please fix typo in the commit message. It should be 290.

File: .luacheckrc (hunk 3 above)
Comment (kyukhin) [reply .luacheckrc:34]: No, it isn't. It is not recursive.

File: .github/actions/environment/action.yml (hunk 4 above)
Comment (kyukhin) [reply .github/actions/environment/action.yml:8]: Done.

File: .luacheckrc [hunk 5]
```diff
@@ -30,7 +30,7 @@ include_files = {
 
 exclude_files = {
     "build/**/*.lua",
-    "test-run/test/test-tarantool/*.test.lua",
+    "test-run/**//*.test.lua",
```
[the remark this answers is not in the document]
Comment (kyukhin) [reply .luacheckrc]: I have no idea in which universe it should be recursive. Your proposal leads to [this](https://github.com/tarantool/tarantool/runs/5621811063?check_suite_focus=true#step:3:511).
See updated branch with failed CI/luacheck.

[the remark this answers is not in the document]
Comment (ylobankov) [reply .luacheckrc]: In our universe it is recursive. Please see https://github.com/tarantool/tarantool/pull/6935#discussion_r830651908 carefully.

**Review by ylobankov — APPROVED** [review]
LGTM

File: .luacheckrc (hunk 5 above)
[the remark this answers is not in the document]
Comment (kyukhin) [reply .luacheckrc]: Wow, seems you're right! Thanks a lot for patience. PR updated.

**Pull request merged**